client/src/client/async.rs

```diff
@@ -0,0 +1,169 @@
+//! This module contains the client for asynchronous communcation.
```

Comment:
should we make a note somewhere that this is not active i.e. does not implement a request method?

Reply:
🤔 
not yet, we haven't published the crate yet and I plan on trying again soon.